qcodes/dataset/sqlite/queries.py

```diff
@@ -292,8 +292,8 @@ def get_parameter_tree_values(conn: ConnectionPlus,
         index is parameter value (first toplevel_param, then other_param_names)
     """
 
-    offset = (start - 1) if start is not None else 0
-    limit = (end - offset) if end is not None else -1
+    offset = max((start - 1), 0) if start is not None else 0
+    limit = max((end - offset), 0) if end is not None else -1
```

Comment:
if my memory is not failing on me, we do have a parametrized test for the start/end, so we could add the negative cases there.

Reply:
Well I ported the parameterized test from get_data to get_parameter_data. That was how I discovered this. I can't find a similar test for get parameter data

Reply:
oh, ok )